master/buildbot/steps/source/git.py

```diff
@@ -204,6 +204,11 @@ def clean(self):
         rc = yield self._cleanSubmodule()
         if rc != RC_SUCCESS:
             raise buildstep.BuildStepFailed
+
+        if self.submodules:
+            rc = yield self._dovccmd(clean_command)
```

Comment:
Test coverage says that we don't have a test for this code path. Could you maybe add it? `test_mode_full_clean` is probably a good candidate to base the effort on.